mountpoint-s3/src/main.rs

```diff
@@ -609,11 +604,11 @@ fn parse_bucket_name(bucket_name: &str) -> anyhow::Result<String> {
     // Actual bucket names must start/end with a letter, but bucket aliases can end with numbers
     // (-s3), so let's just naively check for invalid characters.
     let bucket_regex = Regex::new(r"^[0-9a-zA-Z\-\._]+$").unwrap();
-    // ARN can have `:` and `/` but the prefix is always `arn:`
-    let arn_regex = Regex::new(r#"^arn:aws(-[a-z]+)*:s3(.*):([0-9]+):(.*?)(:(.*?))?$"#).unwrap();
+    // A simple check for AWS ARN
+    let arn_regex = Regex::new(r#"^arn:aws(-[a-z]+)*:.*$"#).unwrap();
     if !bucket_regex.is_match(bucket_name) && !arn_regex.is_match(bucket_name) {
```

Comment:
Let's just do a prefix check.

```suggestion
    if !bucket_regex.is_match(bucket_name) && !bucket_name.starts_with("arn:") {
```